subprojects/core/src/main/java/org/gradle/execution/plan/Node.java

```diff
@@ -223,4 +219,15 @@ public boolean hasHardSuccessor(Node successor) {
 
     @Override
     public abstract String toString();
+
+    public static abstract class Visitor {
+
+        public void visitTask(Task task) {
+        }
+
+        public void visitTransformation(TransformationNodeIdentifier transformation) {
+        }
+
+    }
```

Comment:
This should visit the `Node`s I think for symmetry's sake.

Reply:
You mean this should be `visitTaskNode(TaskNode task)` and `visitTransformNode(TransformNode)`? `TransformNode` is currently usable here because it's in `:dependencyManagement`. 🤔

Reply:
It could be `visit(Node)` too.